app/controllers/summary_controller.rb

```diff
@@ -71,15 +79,22 @@ def page
   end
 
   def sort_params(summary_type)
-    if params['sort'].blank?
-      return [:sentence_start_date, :asc] unless summary_type == :allocated
-
-      return [nil, nil]
-    end
+    return default_sort_params(summary_type) if params['sort'].blank?
```

Comment:
This is a pre-existing problem with this method, but I find it hard to parse the control flow here. It seems like this part isn't well-covered by tests, though, so I would understand not wanting to change it.

That said, my thoughts expressed in code, trying to express the policy of "do something different depending on which params are provided":
```ruby
def sort_params(summary_type)
  sort_params = params['sort'] || ''
  parts = sort_params.split.map { |s| s.downcase.to_sym }

  if parts.blank?
    default_sort_params(summary_type)
  elsif parts.second.blank?
    parts + [:asc]
  else
    parts
  end
end
```

Reply:
Implemented ⬆️ thanks for this